CK.Monitoring/MonitorFileOutputBase.cs

```diff
@@ -169,6 +176,79 @@ public void Dispose()
             Close();
         }
 
+        /// <summary>
+        /// Automatically deletes files that are older than the specified <paramref name="timeSpanToKeep"/>,
+        /// and those beyond that would exceed the total size specified by <paramref name="totalBytesToKeep"/>.
+        /// </summary>
+        /// <param name="m">The monitor to use when logging.</param>
+        /// <param name="timeSpanToKeep">
+        /// The minimum time log files should be kept.
+        /// Log files within this span will not be deleted if they exceed <paramref name="totalBytesToKeep"/>.
+        /// If zero: log files will not be deleted at all.
+        /// </param>
+        /// <param name="totalBytesToKeep">
+        /// The maximum total log file size to keep.
+        /// If zero: all files beyond <paramref name="timeSpanToKeep"/> will be deleted.
+        /// </param>
+        public void RunFileHousekeeping( IActivityMonitor m, TimeSpan timeSpanToKeep, long totalBytesToKeep )
+        {
+            if( timeSpanToKeep == TimeSpan.Zero || _basePath == null ) return;
+
+            DateTime minDate = DateTime.UtcNow - timeSpanToKeep;
+
+            // TODO: What happens if two logfiles have the same date? At worst, conflicting be deleted on the next run
+            // Note: The comparer is a reverse comparer. The most RECENT log file is the FIRST.
+            SortedDictionary<DateTime, FileInfo> sortedLogFiles = new SortedDictionary<DateTime, FileInfo>( Comparer<DateTime>.Create( ( x, y ) => y.CompareTo( x ) ) );
+
+            DirectoryInfo logDirectory = new DirectoryInfo( _basePath );
+            foreach( FileInfo file in logDirectory.EnumerateFiles() )
+            {
+                // Temporary files are "T-" + <date> + _fileNameSuffix + ".tmp" (See OpenNewFile())
+                if( file.Name.EndsWith( ".tmp" ) && file.Name.StartsWith( "T-" ) )
```

Comment:
defensive programming here? Either the file ends with .tmp OR starts with "T-" ?